Assignment 2-Yawei Zhu.Rmd

```diff
@@ -86,15 +88,50 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
-
-
-
+```{r}
+t2<-EDGE%>%select(from,count)%>%group_by(from)%>%summarize(sum(count))
+names(t2)<-c("id","count")
+s2<-EDGE%>%select(to,count)%>%group_by(to)%>%summarize(sum(count))
+names(s2)<-c("id","count")
+INFO<-bind_rows(t2,s2)
+VERTEX<-full_join(VERTEX, INFO, by="id")
+g2<-graph.data.frame(EDGE, vertices=VERTEX, directed=TRUE)
+plot(g2, layout=layout.fruchterman.reingold, edge.arrow.size=0.3,edge.width=EDGE$count, vertex.color=VERTEX$gender, vertex.size=VERTEX$count, vertex.shapes=as.factor(VERTEX$group))
+```
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+library(readr)
+a1 <- read_csv("hudk4050-classes.csv")
+library(tidyr)
+a2 <- unite(a1, name, "First Name", "Last Name")
+a3 <- gather(a2, class, class_number, -name, na.rm = TRUE)
+a4 <- a3 %>% select(name,class_number)
+names(a4) <- c("from", "to")
+a4$count<-1
+
+# create vertex attribute
+stu<-select(a4, from, count)
+stu<-group_by(stu, from)%>%summarise(sum(count))
+stu$group<-"student"
+names(stu)<-c("id","count","group")
+class<-select(a4, to, count)
+class<-group_by(class, to)%>%summarise(sum(count))
+class$group<-"class"
+names(class)<-c("id", "count", "group")
+vertex4050<-bind_rows(stu, class)
```

Comment:
You got part of the way there, you needed to then spread the counts over the classes though, not only summarize them.